pyuvdata/uvdata/ms.py

```diff
@@ -126,11 +754,304 @@ def find_not_ascii(s):
+        history_table.putcol("MESSAGE", message)
+        history_table.putcol("OBJECT_ID", obj_id)
+        history_table.putcol("OBSERVATION_ID", obs_id)
+        history_table.putcol("ORIGIN", origin)
+        history_table.putcol("PRIORITY", priority)
+        history_table.putcol("TIME", times)
+
+        history_table.done()
+
+    def write_ms(
+        self,
+        filepath,
+        force_phase=False,
+        clobber=False,
+        run_check=True,
+        check_extra=True,
+        run_check_acceptability=True,
+        strict_uvw_antpos_check=False,
+    ):
+        """
+        Write a CASA measurement set (MS).
+
+        Parameters
+        ----------
+        filepath : str
+            The MS file path to write to.
+        force_phase : bool
+            Option to automatically phase drift scan data to zenith of the first
+            timestamp.
+        clobber : bool
+            Option to overwrite the file if it already exists.
+        run_check : bool
+            Option to check for the existence and proper shapes of parameters
+            before writing the file.
+        check_extra : bool
+            Option to check optional parameters as well as required ones.
+        run_check_acceptability : bool
+            Option to check acceptable range of the values of parameters before
+            writing the file.
+        strict_uvw_antpos_check : bool
+            Option to raise an error rather than a warning if the check that
+            uvws match antenna positions does not pass.
+
+        """
+        if not casa_present:  # pragma: no cover
+            raise ImportError(no_casa_message) from casa_error
+
+        if run_check:
+            self.check(
+                check_extra=check_extra,
+                run_check_acceptability=run_check_acceptability,
+                strict_uvw_antpos_check=strict_uvw_antpos_check,
+            )
+
+        if os.path.exists(filepath):
+            if clobber:
+                print("File exists; clobbering")
+            else:
+                raise IOError("File exists; skipping")
+
+        # TODO: the phasing requirement can probably go away once we have proper
+        # multi-object support.
+        if self.phase_type == "phased":
+            pass
+        elif self.phase_type == "drift":
+            if force_phase:
+                print(
+                    "The data are in drift mode and do not have a "
+                    "defined phase center. Phasing to zenith of the first "
+                    "timestamp."
+                )
+                phase_time = Time(self.time_array[0], format="jd")
+                self.phase_to_time(phase_time)
+            else:
+                raise ValueError(
+                    "The data are in drift mode. "
+                    "Set force_phase to true to phase the data "
+                    "to zenith of the first timestamp before "
+                    "writing a uvfits file."
```

Comment:
This looks like a copy-paste typo (should be `measurement set` instead of `uvfits`)